marklogic-client-api/src/main/java/com/marklogic/client/impl/RESTServices.java

```diff
@@ -541,37 +553,34 @@ public BufferedMultipleAtomicCallField(String paramName, Stream<String> paramVal
 	}
   }
 	
-  static public class BufferedMultipleNodeCallField extends CallField implements MultipleNodeCallField {
+  static public class BufferedMultipleNodeCallField extends MultipleNodeCallField {
 	  
-		private Stream<BytesHandle> paramValues;
-		public BufferedMultipleNodeCallField(String paramName, Stream<BytesHandle> paramValues) {
+	  private BytesHandle[] paramValues;
+	  public BufferedMultipleNodeCallField(String paramName, Stream<BytesHandle> paramValues) {
 			super(paramName);
-	        this.paramValues = paramValues;
-	    }
-	  
-		@Override
-		public Stream<BytesHandle> getParamValues() {
-			return paramValues;
-		}
-	  
+	        this.paramValues = paramValues.toArray(size -> new BytesHandle[size]);
+	  }
+	  @Override
+	  public Stream<BytesHandle> getParamValues() {
+			return Stream.of(paramValues);
+			}
   }
   
-  static public class BufferedSingleNodeCallField extends CallField implements SingleNodeCallField {
+  static public class BufferedSingleNodeCallField extends SingleNodeCallField {
 	  
-		private Stream<BytesHandle> paramValues;
-		public BufferedSingleNodeCallField(String paramName, Stream<BytesHandle> paramValues) {
+		private BytesHandle paramValue;
+		public BufferedSingleNodeCallField(String paramName, BytesHandle paramValue) {
 			super(paramName);
-	        this.paramValues = paramValues;
+	        this.paramValue = paramValue;
 	    }
 	  
-		public Stream<BytesHandle> getParamValues() {
-			return paramValues;
+		@Override
+		public BytesHandle getParamValue() {
+			return paramValue;
 		}
 
-		@Override
-		public AbstractWriteHandle getParamValue() {
-			// TODO Auto-generated method stub
-			return null;
+		public Stream<AbstractWriteHandle> getParamValues() {
```

Comment:
Given that SingleAtomicCallField doesn't have a stream getter, what is the rationale for SingleNodeCalleField to have a stream getter?